### PR TITLE
ui. sort users and groups

### PR DESCRIPTION
**Pull request by federicoballarini**
Add sorting function to sort users and groups on Cockpit interface.
Refer to https://github.com/NethServer/dev/issues/5969

### CODE REVIEW

File: ui/src/components/system/UsersGroups.vue [hunk 1]
```diff
@@ -2436,6 +2436,27 @@ export default {
...
+        }
+      }
+
+      sortable.sort(function (a, b) {
+        var x = a[0].toLowerCase(),
+            y = b[0].toLowerCase();
+        return x < y ? -1 : x > y ? 1 : 0;
+      });
+      
+      var json = JSON.stringify(sortable);
```
Comment (gsanchietti) [inline ui/src/components/system/UsersGroups.vue]: :fearful: 

You should never manipolate a string to convert one data structure to another.
Please bear in mind that an object doesn't have the concept of order: you need to use and array to preserve the sorting.

Comment (edospadoni) [reply ui/src/components/system/UsersGroups.vue]: Yes, I agree with @gsanchietti. You have to declare a new object like this:
```js
var newUserObj = {};
Object.keys(oldUserObject).sort().forEach(function(key) {
  newUserObj[key] = oldUserObject[key];
});
```

than in `newUserObj` should have the object keys ordered.

Comment (federicoballarini) [reply ui/src/components/system/UsersGroups.vue]: Sorry :D I made changes. Easier than it looks. Thank you so much.

File: ui/src/components/system/UsersGroups.vue [hunk 2]
```diff
@@ -2398,7 +2398,7 @@ export default {
         null,
         function(success) {
           try {
-            success = JSON.parse(success);
+            success = sortList(JSON.parse(success));
```
Comment (edospadoni) [inline ui/src/components/system/UsersGroups.vue]: missing `context` here:

`success = context.sortList(JSON.parse(success));`

File: ui/src/components/system/UsersGroups.vue [hunk 3]
```diff
@@ -2423,7 +2423,7 @@ export default {
...
```
Comment (edospadoni) [inline ui/src/components/system/UsersGroups.vue]: missing `context` here:

`success = context.sortList(JSON.parse(success));`

File: ui/src/components/system/UsersGroups.vue (hunk 2 above)
Comment (federicoballarini) [reply ui/src/components/system/UsersGroups.vue]: Yes, I saw it :) I will update in a minute. My bad

**Pull request merged**